src/components/ResultContainer.tsx

```diff
@@ -58,103 +67,218 @@ function ResultContainer({ response }: { response: QueryResponse | null }) {
     }
   }, [response]);
 
+  function convertURIToLabel(
+    type: string,
+    uri: string | string[] | null
+  ): string | string[] | null {
+    // Handle array of URIs
+    if (Array.isArray(uri)) {
+      return uri.map((singleUri) => convertURIToLabel(type, singleUri)).join(', ');
+    }
+
+    if (!uri) {
+      return uri;
+    }
+
+    switch (type) {
+      case 'sex': {
+        const entry = Object.entries(sexes).find(([, value]) => {
+          const [, id] = value.split(':');
+          return uri.includes(id);
+        });
+        return entry ? entry[0] : uri;
+      }
+
+      case 'sessionType':
+        if (uri.includes('Imaging')) {
+          return 'Imaging';
+        }
+        if (uri.includes('Phenotypic')) {
+          return 'Phenotypic';
+        }
+        return uri;
+
+      case 'diagnosis': {
+        const diagnosis = diagnosisOptions.find((d) => {
+          const [, id] = d.TermURL.split(':');
+          return uri.includes(id);
+        });
+        return diagnosis ? diagnosis.Label : uri;
+      }
+
+      case 'assessment': {
+        const assessment = assessmentOptions.find((a) => {
+          const [, id] = a.TermURL.split(':');
+          return uri.includes(id);
+        });
+        return assessment ? assessment.Label : uri;
+      }
+
+      case 'modality': {
+        const modalityKey = Object.keys(modalities).find((key) => key === uri);
+        return modalityKey ? modalities[modalityKey].label : uri;
+      }
+
+      case 'pipeline':
+        return uri.slice(65);
```

Comment:
:eyes:, I mean, no?
```suggestion
        const segments = uri.split('/');
        return segments[segments.length - 1];
```
or anything that is not tied to a fixed string length.

Reply:
I mean that's how we're truncating it for the pipeline button but I'm good with this implementation as long as we stay consistent.